docs/consuming_with_cmake.rst

```diff
@@ -0,0 +1,91 @@
+.. _consuming_with_cmake:
+
+Consuming with CMake
```

Comment:
I don't understand the purpose of this file. I would avoid the word "consuming" and add an introductory paragraph describing what this is. "Using the oneMKL interfaces in your project with CMake" maybe?

Reply:
The purpose of this is to minimise the friction involved in getting started with using oneMKL in a project with CMake. When I've tried using it, I found that I had to dig though the CMake generated files to work out the targets. Its also easy to accidentally link to Intel(R) oneMKL instead, as in https://github.com/oneapi-src/oneMKL/issues/501. 

I've renamed this and added more of an introduction in [2b9a352](https://github.com/oneapi-src/oneMKL/pull/510/commits/2b9a352c5755b09fc8c395233f1f4393e54fdbe3).